lisa/tools/meson.py

```diff
@@ -30,12 +35,39 @@ def can_install(self) -> bool:
     def _install(self) -> bool:
         posix_os: Posix = cast(Posix, self.node.os)
         # use pip to make sure we install a recent version
-        if (not posix_os.package_exists("meson")) or posix_os.get_package_information(
-            "meson", use_cached=False
-        ) < "0.52.0":
+
+        package_installed = ""
+        package_available = ""
+        # packaged as 'meson' on older systems and 'python3-meson' on newer ones,
+        # since it's actually just a python package.
+        # So check for both
+        for pkg in [
+            "python3-meson",
+            "meson",
+        ]:
+            if posix_os.package_exists(pkg, minimum_version=self._minimum_version):
+                package_installed = pkg
+                break
+            elif posix_os.is_package_in_repo(pkg):
```

Comment:
`elif` to `if` is the same, but it means the two `if` blocks are actually independent.

Reply:
Can you be more specific with your request from me?

Reply:
```suggestion
            if posix_os.is_package_in_repo(pkg):
```

Reply:
Right. So making _just_ that change will introduce an unintended side effect of updating the package before we check the version of the currently installed package. Let me see if I can make the decisions more clear in the code. I was _intentionally_ not installing an updated version from the package manager if one was already installed because it was convenient for my meson use cases... But really it would be better to write for the general case; even though it will be a little uglier and use another round trip or two.

Reply:
It wasn't so bad to rework! Testing the more commented, reworked version now